testkit/jvm/src/main/scala/zio/test/ZIOTest.scala

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2019 John A. De Goes and the ZIO Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package zio.test
+
+import zio._
+import zio.internal.{ Platform, PlatformLive }
+
+abstract class ZIOTest[R, E] { self =>
+
+  /**
+   * Construct your environment here.
+   */
+  val environment: Managed[Nothing, R]
+
+  /**
+   * Place all your tests here.
+   */
+  val tests: ZSpec[R, E, String]
```

Comment:
This should be a `def` to allow accumulation through mutable DSLs

Reply:
Done... but eww! 😉